OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -42,22 +46,21 @@ public override Activity Tick(Actor self)
 					return this;
 			}
 
-			if (!target.IsValidFor(self))
-				return NextActivity;
-
-			if (IsCanceling)
+			if (IsCanceling || target.Type == TargetType.Invalid)
 			{
 				aircraft.RemoveInfluence();
 				return NextActivity;
```

Comment:
This returns without putting the actor back into a consistent state - stopping the helicopter or losing the target while the land is half done will leave the helicopter sitting at an intermediate altitude.

This is already the case on bleed and I think may already be solved in #16509, so we don't need to address this here.